### PR TITLE
refactor Magazine template

### PR DESCRIPTION
**Pull request by jonatan-lledo-netcentric**
Fix https://github.com/hlxsites/vg-volvotrucks-us/issues/683

I added `tags.xlsx` as config file to be used to add all tags to later also be used to fill the 3 filters (categories, topics & trucks like VT does).

this JSON file is located in `/magazine/articles` folder.

In this refactor I check if an `article:tag` metadata is in the JSON file in one of the 2 lists to be added in the Hero.

Also I refactor `recommended` block to handle the new category's origin and I left a `TODO` in `recent-articles` block because the `magazine-articles.json` file is not updated yet

So because the magazine page shares the same code across more than 1 block I moved the proper functions to `common.js`

#

Test URLs:
- Before: https://main--vg-macktrucks-com--hlxsites.hlx.page/magazine/articles/2023/august/desert-ready-mix-finds-success-with-help-from-mack-financial-services
- After: https://683-search-metadata-update--vg-macktrucks-com--hlxsites.hlx.page/magazine/articles/2023/august/desert-ready-mix-finds-success-with-help-from-mack-financial-services

### CODE REVIEW

**Review by amglebihan — APPROVED** [review]

**Review by alexiscoelho — APPROVED** [review]

File: templates/magazine/magazine.js [hunk 1]
```diff
@@ -88,7 +89,30 @@ async function buildShareSection() {
...
+
+  const matchingTags = [...articleTags]
+    .filter((tag) => tags.includes(tag.content))
+    .map((tag) => tag.content);
+  return matchingTags && matchingTags?.length > 0 ? matchingTags.join(', ') : '';
+}
+
 export default async function decorate(doc) {
+  const articleTags = doc.head.querySelectorAll('meta[property="article:tag"]') || [];
+  const tagItems = await getJsonFromUrl('/magazine/articles/tags.json');
```
Comment (alexiscoelho) [inline templates/magazine/magazine.js:110]: Check if tagItems is truthy before moving forward

File: templates/magazine/magazine.js [hunk 2]
```diff
@@ -88,7 +89,30 @@ async function buildShareSection() {
...
+  const categories = tagItems && getValuesfromObjectsArray(tagItems.categories.data);
```
Comment (alexiscoelho) [inline templates/magazine/magazine.js:111]: tagItems && tagItems.categories &&

File: templates/magazine/magazine.js [hunk 3]
```diff
@@ -88,7 +89,30 @@ async function buildShareSection() {
...
+  const trucks = tagItems && getValuesfromObjectsArray(tagItems.trucks.data);
```
Comment (alexiscoelho) [inline templates/magazine/magazine.js:112]: tagItems && tagItems.trucks &&

**Review by alexiscoelho — CHANGES_REQUESTED** [review]

**Review by alexiscoelho — APPROVED** [review]

File: templates/magazine/magazine.js (hunk 1 above)
Comment (jonatan-lledo-netcentric) [reply templates/magazine/magazine.js]: done

File: templates/magazine/magazine.js (hunk 2 above)
Comment (jonatan-lledo-netcentric) [reply templates/magazine/magazine.js]: done

File: templates/magazine/magazine.js (hunk 3 above)
Comment (jonatan-lledo-netcentric) [reply templates/magazine/magazine.js]: done

File: scripts/common.js [hunk 4]
```diff
@@ -586,3 +586,47 @@ export const clearElementAttributes = (element) => {
 
   return element;
 };
+
+// magazine common functions
+/**
+ * Extracts the values from an array of objects and returns an array of values
+ * example: [{ key: 'value' }] => ['value']
+ * @param {array} array - An array of objects
```
Comment (alexiscoelho) [inline scripts/common.js:594]: Should be: `{Array}`
And not: `{array}`
Same for the other array docs

File: scripts/common.js [hunk 5]
```diff
@@ -586,3 +586,47 @@ export const clearElementAttributes = (element) => {
...
+
+/**
+ * Get the article tags from the JSON file and the article tags from the document
+ * and return the matching tags
+ * @param {string} tagType - The type of tag to get such as 'categories', 'topics' or 'trucks'
+ * @returns {string} A string of matching tags
+ */
+export async function getArticleTags(tagType) {
+  const articleTags = document.head.querySelectorAll('meta[property="article:tag"]') || [];
+  const tagItems = await getJsonFromUrl('/magazine/articles/tags.json') || null;
```
Comment (alexiscoelho) [inline scripts/common.js:628]: getJsonFromUrl already returns null in case of failure

**Review by alexiscoelho — APPROVED** [review]

File: scripts/common.js (hunk 4 above)
Comment (jonatan-lledo-netcentric) [reply scripts/common.js]: done

File: scripts/common.js (hunk 5 above)
Comment (jonatan-lledo-netcentric) [reply scripts/common.js]: done

**Review by santi-homps — APPROVED** [review]

**Pull request merged**